tools/src/bin/output-file.rs

```diff
@@ -57,6 +54,110 @@ fn get_bugzilla_component<'a>(
     Some((product, component))
 }
 
+/// Information about expected failures/problems for specific web platform
+/// tests.
+struct WPTExpectationInfo {
+    /// The condition strings and related bugs that disable this test in its
+    /// entirety.
+    disabling_conditions: Vec<(String, String)>,
+    /// The number of `_subtests` that were disabled or conditionally disabled
+    disabled_subtests_count: i64,
+}
+
+/// Information from `test-info-all-tests.json` which knows about files that the
+/// test manifests know about.
+struct TestInfo {
+    failed_runs: i64,
+    skip_if: Option<String>,
+    skipped_runs: i64,
+    total_run_time_secs: f64,
+    /// "total runs" less "skipped runs"
+    unskipped_runs: i64,
+    /// For web platform tests with expected failures/problems, the info about
+    /// that.  Tests that are expected to succeed will have None here.
+    wpt_expectation_info: Option<WPTExpectationInfo>,
+}
+
+fn read_test_info_from_file_info(per_file_info: Option<&json::Object>) -> Option<TestInfo> {
+    let obj = per_file_info?.get("testInfo")?.as_object()?;
+
+    let failed_runs = match obj.get("failed runs") {
+        Some(json) => json.as_i64().unwrap(),
+        _ => 0
+    };
+    let skip_if = match obj.get("skip-if") {
+        Some(json) => Some(json.as_string().unwrap().to_string()),
+        _ => None
+    };
+    let skipped_runs = match obj.get("skipped runs") {
+        Some(json) => json.as_i64().unwrap(),
+        _ => 0
+    };
+    let total_run_time_secs= match obj.get("total run time, seconds") {
+        Some(json) => json.as_f64().unwrap(),
+        _ => 0.0
+    };
+    let total_runs = match obj.get("total runs") {
+        Some(json) => json.as_i64().unwrap(),
+        _ => 0
+    };
+
+    let wpt_expectation_info = match per_file_info?.get("wptInfo") {
+        Some(Json::Object(obj)) => {
+            let disabling_conditions = match obj.get("disabled") {
+                Some(Json::Array(arr)) => arr.iter().filter_map(|cond| {
+                    // cond itself should be a 2-element array where the first
+                    // element is a null or the condition string and the 2nd is
+                    // the bug link.
+                    match cond.as_array().unwrap_or(&vec![]).as_slice() {
+                        // null means there was no condition, it's always disabled.
+                        [Json::Null, Json::String(b)] => Some(("ALWAYS".to_string(), b.to_string())),
+                        [Json::String(a), Json::String(b)] => Some((a.to_string(), b.to_string())),
+                        // I guess this is just covering up our failures?  I'm
+                        // sorta tired of this patch though, so... cover up our
+                        // failures.
+                        _ => None,
```

Comment:
Covering up is fine, but maybe add a `warn!` of some sort here so that if it happens we'll get a notification about it.